### DISCUSSION TITLE
Support nameof() expression

### DISCUSSION THREAD

**Discussion by The-Futurist** [dotnet/csharplang#8664] [Language Ideas]
Sometimes we have code that looks like this:

```      
        public const string resource_link_title = nameof(resource_link_title);
        public const string resource_link_id = nameof(resource_link_id);
        public const string resource_link_description = nameof(resource_link_description);
```

I'm suggesting we allow this form of `nameof` in cases where the assignment target is a string variable:

```        
        public const string resource_link_title = nameof();
        public const string resource_link_id = nameof();
        public const string resource_link_description = nameof();
```

and so the `nameof()` is treated as if it were the explicit form, not necessarily just for declarations but in any case where we are assigning to a string variable.

**Comment by siegfriedpammer** [comment]
Seems like a very niche use case for a language feature considering the high cost of implementing it in the whole ecosystem.

I'm pretty sure you can achieve this with Source Generators today...

**Comment by jnm2** [comment]
I always wonder with these: if the variable were to be renamed, would you really want the string to change? Wouldn't that cause an incompatibility with deserialization, or something, or is there no "outside world" to worry about? Why use `nameof` at all here?

**Comment by HaloFour** [comment]
How about `partial const`? 😄

**Reply by CyrusNajmabadi** [reply]
I didn't actually have an issue with that.  With partial props, we've made the goals and designs pretty clear in this space.
